sdk/utils.go

```diff
@@ -190,3 +192,132 @@ func logStartupInfo() {
 	}
 	log.Info("--------------------------------")
 }
+
+// RedactPasswords redacts map fields containing key substring "pass"
+// (case-insensitive) in Marshaled json string s, and returns a string where
+// those field values are emitted as REDACTED.
+func RedactPasswords(s string) (output string, err error) {
+
+	// Unmarshal json string to structure.
+	var x interface{}
+	err = json.Unmarshal([]byte(s), &x)
+	if err != nil {
+		return
+	}
+
+	// We only need to be concerned with the following types.
+	// If someone has a magic string that happens to be a password, we cannot
+	// help you.
+	switch v := x.(type) {
+	case map[string]interface{}:
+		err = traverseMap(x.(map[string]interface{}))
+	case []interface{}:
+		err = traverseSlice(x.([]interface{}))
+	default:
+		err = fmt.Errorf("Unsupported type: %v\n", v)
+	}
+
+	if err != nil {
+		fmt.Printf("Error traversing: %v\n", err)
+		return
+	}
+
+	// Marshal to bytes. Return string.
+	outputTemp, err := json.Marshal(x)
+	if err != nil {
+		return
+	}
+	return string(outputTemp), nil
+}
+
+// traverseMap iterates through all keys and values in a map[string]interface{}.
+// If it finds a nested map[string]interface{} we recurse into it.
+func traverseMap(m map[string]interface{}) (err error) {
```

Comment:
These functions look good to me, but I'd like to see additional test cases for them since I don't think the existing one hits  all of the cases, particularly with a nested slice. This could be done in a later PR as well.

Reply:
It's not easy to see, but it definitely hits the nested slice in these brackets: \"Config\":[ ... \"v3\"}]}

This is the slice of maps:
https://github.com/vapor-ware/synse-sdk/blob/b79b1b2772a6ae890d53661ae5aa6f1691cdf887/sdk/plugin.go#L495